kolibri/utils/server.py

```diff
@@ -395,6 +387,24 @@ def handler(signum, frame):
 
     # Start the server engine (Option 1 *and* 2)
     cherrypy.engine.start()
+
+    cherrypy.server.wait()
+    bind_addr, bind_port = cherrypy.server.bound_addr
+
+    # Write the PID file again, in case the port number has changed
+    _write_pid_file(PID_FILE, port=bind_port)
+
+    # Register the Kolibri zeroconf service so it will be discoverable on the network
+    from kolibri.core.discovery.utils.network.search import (
+        register_zeroconf_service,
+    )
+
+    register_zeroconf_service(port=bind_port)
```

Comment:
Hi LE friends! Just passing by to see what new stuff is happening with Kolibri :heart: 

I was wondering a bit about this change since CherryPy isn't the only HTTP server that can be in front of Kolibri. So if the Zeroconf service should be started together with the other background services, then the old location is better?

Reply:
Hrm - that's a good point, this would mess up use of zeroconf behind an nginx/uwsgi deployment for example. Good catch.

There is the slight advantage here that in the cherrypy instance it only registers zeroconf once the port is actually available, so we may need to go for a two pronged approach.